pkg/controller/testutil/test_utils.go

```diff
@@ -480,3 +487,17 @@ func GetZones(nodeHandler *FakeNodeHandler) []string {
 func CreateZoneID(region, zone string) string {
 	return region + ":\x00:" + zone
 }
+
+// GetKey is a helper function used by controllers unit tests to get the
+// key for a given kubernetes resource.
+func GetKey(obj interface{}, t *testing.T) string {
+	val := reflect.ValueOf(obj).Elem()
+	name := val.FieldByName("Name").String()
+	kind := val.FieldByName("Kind").String()
```

Comment:
`obj` in the test doesn't always have its `name` or `kind` set because it doesn't go through API server for real.

Reply:
right but in all places its used, it was called with 
```
if err != nil {
-		t.Errorf("Unexpected error getting key for ds %v: %v", ds.Name, err)
-	}
```
this means it always had Name anyways . Do you want me to do additional validation ?